tests/cases/unittests/session.ts

```diff
@@ -0,0 +1,459 @@
+				let req : protocol.FileRequest = {
+					command: CommandNames.Open,
+					seq: 0,
+					type: 'command',
+					arguments: {
+						file: undefined
+					}
+				};
+
+				expect(() => session.executeCommand(req)).to.throw();
+			});
+			it('should output an error response when a command does not exist', () => {
+				let req : protocol.Request = {
+					command: 'foobar',
+					seq: 0,
+					type: 'command'
+				};
+
+				session.executeCommand(req);
+
+				expect(lastSent).to.deep.equal(<protocol.Response>{
+					command: CommandNames.Unknown,
+					type: 'response',
+					seq: 0,
+					message: 'Unrecognized JSON command: foobar',
+					request_seq: 0,
+					success: false
+				});
+			});
+			it('should return a tuple containing the response and if a response is required on success', () => {
+				let req : protocol.ConfigureRequest = {
+					command: CommandNames.Configure,
+					seq: 0,
+					type: 'command',
+					arguments: {
+						hostInfo: 'unit test',
+						formatOptions: {
+							newLineCharacter: '`n'
+						}
+					}
+				};
+
+				expect(session.executeCommand(req)).to.deep.equal({
+					responseRequired: false
+				});
+				expect(lastSent).to.deep.equal({
+					command: CommandNames.Configure,
+					type: 'response',
+					success: true,
+					request_seq: 0,
+					seq: 0,
+					body: undefined
+				});
+			});
+		});
+
+		describe('onMessage', () => {
+			it('should not throw when commands are executed with invalid arguments', () => {
+				let i = 0;
+				for (name in CommandNames) {
+					if (!Object.prototype.hasOwnProperty.call(CommandNames, name)) {
+						continue;
+					}
+					let req : protocol.Request = {
+						command: name,
+						seq: i++,
+						type: 'command'
+					};
+					session.onMessage(JSON.stringify(req));
+					req.seq += 2;
+					req.arguments = {};
+					session.onMessage(JSON.stringify(req));
+					req.seq += 2;
+					req.arguments = null;
+					session.onMessage(JSON.stringify(req));
+				}
+			});
+			it('should output the response for a correctly handled message', () => {
+				let req : protocol.ConfigureRequest = {
+					command: CommandNames.Configure,
+					seq: 0,
+					type: 'command',
+					arguments: {
+						hostInfo: 'unit test',
+						formatOptions: {
+							newLineCharacter: '`n'
+						}
+					}
+				};
+
+				session.onMessage(JSON.stringify(req));
+
+				expect(lastSent).to.deep.equal(<protocol.ConfigureResponse>{
+					command: CommandNames.Configure,
+					type: 'response',
+					success: true,
+					request_seq: 0,
+					seq: 0,
+					body: undefined
+				});
+			});
+		});
+
+		describe('exit', () => {
+			it('is a noop which can be handled by subclasses', () => {
+				session.exit(); // Does nothing, should keep running tests
+				expect(session).to.exist;
+			});
+		});
+
+		describe('send', () => {
+			it('is an overrideable handle which sends protocol messages over the wire', () => {
+				let msg = {seq: 0, type: 'none'},
+				strmsg = JSON.stringify(msg),

```

Comment:
Use a separate variable statement, and I'd just do `let msg = JSON.stringify({seq: 0, type: "none" });`